daemon/logger/jsonfilelog/jsonfilelog.go

```diff
@@ -124,72 +124,85 @@ func (l *JSONFileLogger) Log(msg *logger.Message) error {
 }
 
 func writeLog(l *JSONFileLogger) (int64, error) {
+
+	if err := checkCapacityAndRotate(l); err != nil {
+		return -1, err
+	}
+
+	i, err := l.buf.WriteTo(l.f)
+	if err != nil {
+		l.buf.Reset()
```

Comment:
This is not necessary. `buf.WriteTo()` already does this: https://golang.org/src/bytes/buffer.go?s=7168:7183#L202